# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: gfx/2d/Types.h
# Commit: e178aa7d969e
# Full Hash: e178aa7d969e31550222e903b8844f80732d73bd
# Author: Lee Salzman <lsalzman@mozilla.com>
# Date: 2022-02-12 09:47:43
# Regressor Bug: 1754130
# File Overlap Count: 2
# Description:
#   Bug 1754130 - Implement SourceSurfaceWebgl for faster snapshots of DrawTargetWebgl. r=aosmond
#   
#   Now that each DrawTargetWebgl shares the same WebGL context, we can efficiently draw snapshots
#   of one DrawTargetWebgl to another without requiring any readback or error-prone driver-provided
#   shared context/resource mechanism. We just need to simply pass the WebGL texture from one target
# ==============================================================================

diff -r acbba09f0bce -r e178aa7d969e gfx/2d/Types.h
--- a/gfx/2d/Types.h	Fri Feb 11 19:49:55 2022 +0000
+++ b/gfx/2d/Types.h	Fri Feb 11 19:49:55 2022 +0000
@@ -41,6 +41,7 @@
   DATA_SHARED_WRAPPER,    /* Shared memory mapped in from another process */
   BLOB_IMAGE,             /* Recorded blob image */
   DATA_MAPPED,            /* Data surface wrapping a ScopedMap */
+  WEBGL,                  /* Surface wrapping a DrawTargetWebgl texture */
 };
 
 enum class SurfaceFormat : int8_t {
